src/shared/components/CloudUpgradeNavBanner.tsx

```diff
@@ -62,7 +63,22 @@ const CloudUpgradeNavBanner: FC<StateProps> = ({inView}) => {
             href={`${CLOUD_URL}${CLOUD_CHECKOUT_PATH}`}
             target="_self"
           >
-            <Icon glyph={IconFont.Upgrade} />
+            <GoogleOptimizeExperiment
+              experimentID="e44rY7GjQN-ASmGeWLs_pA"
```

Comment:
Love how easy this is to read. This is just for A/B test with one variant, but it looks like we could add multiple variants too for a multi-variate test. Niiice.

Reply:
I wanted to start with a simple test; this one is 2 variants plus the original.

Reply:
so this will work for this PR and experiment, but in general, having a key or an id hard coded into code like this is an indicator we need a little more abstraction. right now, it'll only work for the specific experiment id we provide in code, which means it'll only work in prod for a specific container, right? if we abstract this `experimentId` into an environment variable, we can inject the correct value per environment (e.g. tools, staging, prod), which would let us test in non-production environments (or any arbitrary environment).

using component props for experiment handling is fine for a couple of variants and for simple components, but it might get unwieldy really quickly for complex components or component trees and hierarchies.

Reply:
Let's chat a bit more about how we can abstract this in the future.